overlord/assertstate/assertstate.go

```diff
@@ -385,7 +386,80 @@ func AutoRefreshAssertions(s *state.State, userID int) error {
 	if err := RefreshSnapDeclarations(s, userID, opts); err != nil {
 		return err
 	}
-	return RefreshValidationSetAssertions(s, userID, opts)
+	if err := RefreshValidationSetAssertions(s, userID, opts); err != nil {
+		return err
+	}
+
+	return autoRefreshRegistryAssertions(s, userID, opts)
+}
+
+// autoRefreshRegistryAssertions fetches the newest revision of all stored
+// registry assertions.
+func autoRefreshRegistryAssertions(st *state.State, userID int, opts *RefreshAssertionsOptions) error {
+	db := cachedDB(st)
+	regAsserts, err := db.FindMany(asserts.RegistryType, nil)
+	if err != nil {
+		if errors.Is(err, &asserts.NotFoundError{}) {
+			return nil
+		}
+		return err
+	}
+
+	var registries []*registry.Registry
+	for _, regAs := range regAsserts {
+		reg := regAs.(*asserts.Registry).Registry()
+		registries = append(registries, reg)
+	}
+
+	return refreshRegistryAssertions(st, registries, userID, opts)
+}
+
+// refreshRegistryAssertions fetches new revisions for the registry assertions
+// referenced by the provided registries. It attempts a bulk refresh and if that
+// fails, it falls back to fetching the assertions one by one.
+func refreshRegistryAssertions(st *state.State, registries []*registry.Registry, userID int, opts *RefreshAssertionsOptions) error {
+	if opts == nil {
+		opts = &RefreshAssertionsOptions{}
+	}
+
+	deviceCtx, err := snapstate.DevicePastSeeding(st, nil)
+	if err != nil {
+		return err
+	}
+
+	err = bulkRefreshRegistries(st, registries, userID, deviceCtx, opts)
+	if err == nil {
+		return nil
+	}
+
+	if _, ok := err.(*bulkAssertionFallbackError); !ok {
+		// not an error that indicates the server rejecting/failing
+		// the bulk request itself
+		return err
+	}
+	logger.Noticef("bulk refresh of registry assertions failed, falling back to one-by-one assertion fetching: %v", err)
+
+	modelAs := deviceCtx.Model()
+	return doFetch(st, userID, deviceCtx, nil, func(f asserts.Fetcher) error {
+		for _, registry := range registries {
+			if err := snapasserts.FetchRegistry(f, registry.Account, registry.Name); err != nil {
+				return err
+			}
+		}
+
+		if modelAs.Store() != "" {
+			err := snapasserts.FetchStore(f, modelAs.Store())
```

Comment:
why is this done for registries, a left over of copying something else here?

Reply:
Yes, I've removed this now